compiler/p/codegen/ControlFlowEvaluator.cpp

```diff
@@ -3001,6 +3001,7 @@ static void lookupScheme3(TR::Node *node, bool unbalanced, TR::CodeGenerator *cg
          TR_PPCTableOfConstants::setTOCSlot(offset, address);
          if (offset<LOWER_IMMED||offset>UPPER_IMMED)
             {
+            TR_ASSERT_FATAL(0x00008000 != cg->hiValue(offset), "TOC offset (0x%x) is unexpectedly high. Can not encode upper 16 bits into an addis instruction.", offset);
```

Comment:
I think this ASSERT and similar below are not needed. Since we are passing int32_t to the instruction the assert inside the binary encoding (the subject of this PR) will fail.

Reply:
While these aren't strictly needed, I requested they be added in https://github.com/eclipse/omr/pull/5213#discussion_r430041047 to provide more information about the problem (the full pTOC offset), to cause the assertion failure earlier, and to make it clear that this is something that was considered but deemed to be an impossible situation.

That being said, can these be switched to use `TR_ASSERT_FATAL_WITH_NODE` so that we can get node context if they fire?

Reply:
I can change it to use `TR_ASSERT_FATAL_WITH_NODE`.